samples/swift/uidemo/Samples/FIRStorageViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+
+import FirebaseStorageUI
+
+class FIRStorageViewController: UIViewController {
+
+  @IBOutlet private var imageView: UIImageView!
+  @IBOutlet private var textField: UITextField!
+
+  private var storageRef = FIRStorageReference().child("sample.jpg")
+
+  override func viewDidAppear(animated: Bool) {
+    super.viewDidAppear(animated)
+    self.textField.autocorrectionType = .No
+    self.imageView.contentMode = .ScaleAspectFit
+
+    // Notification boilerplate to handle keyboard appearance/disappearance
+    NSNotificationCenter.defaultCenter().addObserver(self,
+                                                     selector: #selector(keyboardWillShow),
+                                                     name: UIKeyboardWillShowNotification,
+                                                     object: nil)
+    NSNotificationCenter.defaultCenter().addObserver(self,
+                                                     selector: #selector(keyboardWillHide),
+                                                     name: UIKeyboardWillHideNotification,
+                                                     object: nil)
+  }
+
+  @IBAction private func loadButtonPressed(sender: AnyObject) {
+    self.imageView.image = nil
+    guard let text = self.textField.text else { return }
+    guard let url = NSURL(string: text) else { return }
+    guard let components = url.pathComponents else { return }
+
+    self.storageRef = FIRStorageReference()
+
+    for component in components {
+      self.storageRef = self.storageRef.child(component)
+    }
+
+    self.imageView.sd_setImageWithStorageReference(self.storageRef)
+  }
+
+  // MARK: Keyboard boilerplate
```

Comment:
Not a requirement now, but I assume this is shared code across apps, so it should be in a single place.